src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -239,7 +228,7 @@ public static void Unregister()
             if (!IsRegistered)
             {
                 var error = $"{typeof(MSBuildLocator)}.{nameof(Unregister)} was called, but no MSBuild instance is registered." + Environment.NewLine;
-                if (numResolvedAssemblies == 0)
+                if (!registerCalled)
                 {
                     error += $"Ensure that {nameof(RegisterInstance)}, {nameof(RegisterMSBuildPath)}, or {nameof(RegisterDefaults)} is called before calling this method.";
```

Comment:
I think this whole error system might be kinda useless now--we no longer really have the concept of `Unregister` since the assemblies we might resolve is now unbounded. This change is OK but I think it's worth thinking of just rewriting this whole thing to do nothing.